### DISCUSSION TITLE
🆕 Discussions/Actions integration and Detail view "what's new" Private Beta

### DISCUSSION THREAD

**Discussion by evi-liu** [community/community#4173] [Discussions]
👋 Hi folks! We have two new exciting Discussions private beta features available for testing.

## Discussions integration with GitHub Actions
You can now trigger GitHub Actions with `Discussion` and `DiscussionComments` events. See [these docs](https://docs.github.com/en/actions/reference/events-that-trigger-workflows) for existing information on how to trigger workflows using these events. 

![image](https://user-images.githubusercontent.com/8809849/121955221-89362880-cd14-11eb-940d-a1802005ab71.png)

## Detail view "what's new"
We've heard your feedback that it's difficult to follow along with new comments in longer discussions. So, we're excited to bring some new features to help direct your attention on new information. With this change, you'll be able to jump directly to the new comments AND see when new comments have been added to a thread.

_Jump down to see new comments_
![image](https://user-images.githubusercontent.com/8809849/121956495-0dd57680-cd16-11eb-97a6-61bff0ab679d.png)

_See when there are new comments in the thread_
![image](https://user-images.githubusercontent.com/8809849/121956711-4bd29a80-cd16-11eb-8200-6d2d60ba7b2a.png)

## How do I get started?
If you'd like to try out either of these new features, comment below with: 

1. which of these features (or both) you want to try out AND
2. which repository you want to enable them on.

The private beta is limited to the first 15 responders.

**Comment by laymonage** [comment]
Hey, thanks for the updates!

I'd like to try both features on https://github.com/laymonage/giscus, please! Also, if it's not too much to ask, on https://github.com/laymonage/discussions-playground as well (it's where I try stuff).

**Reply by evi-liu** [reply]
Both repos have been flagged in for Actions and I flagged in your username for the "what's new" feature :)

**Comment by wohali** [comment]
We'd love to try just the "what's new" feature on `apache/couchdb`. Thanks!

**Reply by evi-liu** [reply]
That one is actually a per-user flag, so I flagged in your username. Let me know if there are other folks who'd like to try it out.

**Comment by Andre601** [comment]
One question I have is, are the API endpoints publicly available, or will the Action itself also need to be added for the beta to have access?

I talk about Actions such as [`peaceiris/actions-label-commenter`](https://github.com/peaceiris/actions-label-commenter) which would benefit from supporting this.

**Reply by evi-liu** [reply]
I hope I'm interpreting your question correctly, but this will make it so that you can trigger Actions on `Discussion` and `DiscussionComment` events from our [Discussions webhooks](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#discussion). You shouldn't have to add your Action to the beta once I flag in your repository.

**Reply by peaceiris** [reply]
@evi-liu We want to test the action by ourselves! Please give us the beta flag (both)!

**Reply by evi-liu** [reply]
Got it, the repo has been flagged in and I flagged in both your usernames for the "what's new" feature.

**Reply by peaceiris** [reply]
I have found that we have no way to manage discussions via REST API, so I have opened a feature request at [Discussions REST API and clients · Issue #83 · octokit/rest.js](https://github.com/octokit/rest.js/issues/83).

**Comment by ghost** [comment]
https://codeql.github.com/docs/codeql-overview/

**Reply by andrewdmaclean** [reply]
Hey @evi-liu 👋 I would love to try out both of the new features on [https://github.com/battlesnakeofficial/feedback](https://github.com/battlesnakeofficial/feedback)

**Reply by evi-liu** [reply]
@andrewdmaclean I just flagged in the repo for Actions and flagged your username for the "what's new" feature :). Happy testing!

**Comment by Griatch** [comment]
@evi-liu I made one of the early reports about the trouble to follow along in a thread, so I'd be happy to test out the "what's new" feature. Not sure I need the new actions, but I can check them out too, why not :) 

https://github.com/evennia/evennia/discussions

**Comment by Andre601** [comment]
To give some feedback, I feel like the "New" line could/should be changed to a green colour.
The orange just doesn't feel that appealing to me and more like a "oh no! Something isn't good here!" colour than a info colour.

I use the dark theme (NOT dimmed dark theme)

**Reply by evi-liu** [reply]
Thanks for the feedback! We decided to shift away from orange and instead try out blue. Let me know what you think.

**Reply by Andre601** [reply]
The blue doesn't look bad.
Still think a green-ish colour would've been better, but the blue is already a decent change... Just "blends in" a bit too much with the current UI/UX(?) of GitHub which already uses a lot of blue everywhere.

On this screenshot alone can you see at least 4 different things in a blue colour, so maybe change those or maybe change the shade of the "new" indicator.
![image](https://user-images.githubusercontent.com/11576465/133691111-8f85eb21-f120-4976-9395-de19f9d67bec.png)

**Comment by denelon** [comment]
@evi-liu I'd love to get Detail view "what's new" on https://github.com/microsoft/winget-cli. If you don't mind multiple repositories https://github.com/microsoft/winget-pkgs would be great as well.

**Comment by thanhcsf** [comment]
I'd like to try both features on https://github.com/mlbvn/handson-ml2-vn. Thanks!

**Comment by evi-liu** [comment]
👋 Hi folks, apologies on the delayed response as our team attended to higher priority items. You all should now have the ability to [trigger Discussion events on GitHub Actions](https://github.blog/changelog/2021-06-29-github-discussions-events-available-on-github-actions/). 

As for the "what's new" feature, I've flagged in all the users on this thread to test it out. Please do let me know in this thread if you have any further feedback.

**Comment by Relequestual** [comment]
Hello @evi-liu. Several members of the JSON Schema community are finding it hard to get to find new comments in Discussions now.
Please can you enable the "what's new" preview for https://github.com/json-schema-org/community ?
Thanks.
